src/NzbDrone.Core/Datastore/Migration/198_parse_titles_from_existing_subtitle_files.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections.Generic;
+using System.Data;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Dapper;
+using FluentMigrator;
+using NzbDrone.Core.Datastore.Migration.Framework;
+using NzbDrone.Core.MediaFiles;
+using NzbDrone.Core.MediaFiles.EpisodeImport.Aggregation.Aggregators;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(198)]
+    public class parse_title_from_existing_subtitle_files : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            Alter.Table("SubtitleFiles").AddColumn("Title").AsString().Nullable();
+            Alter.Table("SubtitleFiles").AddColumn("Copy").AsInt32().WithDefaultValue(0);
+            Execute.WithConnection(UpdateTitles);
+        }
+
+        private void UpdateTitles(IDbConnection conn, IDbTransaction tran)
+        {
+            var updates = new List<object>();
+
+            using (var cmd = conn.CreateCommand())
+            {
+                cmd.Transaction = tran;
+                cmd.CommandText = "SELECT \"Id\", \"RelativePath\", \"EpisodeFileId\", \"Language\", \"LanguageTags\" FROM \"SubtitleFiles\"";
+
+                using var reader = cmd.ExecuteReader();
+                while (reader.Read())
+                {
+                    var id = reader.GetInt32(0);
+                    var relativePath = reader.GetString(1);
+                    var episodeFileId = reader.GetInt32(2);
+
+                    var episodeFile = conn.QuerySingle<EpisodeFile>("SELECT * FROM \"EpisodeFiles\" WHERE \"Id\" = @Id", new { Id = episodeFileId });
```

Comment:
We should avoid a separate query for this, maybe via a join getting only the pieces of information that this needs.